2_observations/src/data_munge_functions.R

```diff
@@ -85,22 +80,54 @@ munge_split_temp_dat <- function(dat_ind,
                                  prioritize_nwis_sites = TRUE) {
 
   drb_dat <- readRDS(sc_retrieve(dat_ind, 'getters.yml'))
+  # first find sites that have the same site_id but have a sub-location
+  # this is restricted to NWIS sites that returned multiple columns per site
+  # can be multiple sensors (concurrently measuring at different locations), or could represent
+  # different sensors through time. Some sensors we do not want to keep (piezometers)
+
+  # first handle multiple sub_locations
+  # that are causing >1 obs per site_id-date
+  sub_location_res <- drb_dat %>%
+    filter(!grepl('piezometer', sub_location, ignore.case = TRUE)) %>%
+    filter(!is.na(sub_location)) %>%
+    group_by(site_id, sub_location) %>%
+    mutate(n_sub = n()) %>% ungroup() %>%
```

Comment:
I like this approach for handling the site-dates with multiple sub-locations. Do I have it right that `n_sub` here is the number of days that each sub_location has recorded temperature data? For future reference, perhaps we could add a comment that for site_id-date with >1 observations coming from multiple sub_locations, we preferentially select the sub_location with the greater number of total observation-days.